components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -5637,8 +5637,14 @@ public Map<String, Object> searchPaginatedAPIs(String searchQuery, String organi
                 List<Object> apiList = new ArrayList<>();
                 for (DevPortalAPIInfo devPortalAPIInfo : list) {
                     API mappedAPI = APIMapper.INSTANCE.toApi(devPortalAPIInfo);
-                    mappedAPI.setRating(APIUtil.getAverageRating(mappedAPI.getUuid()));
-                    apiList.add(mappedAPI);
+                    try {
+                        mappedAPI.setRating(APIUtil.getAverageRating(mappedAPI.getUuid()));
+                        apiList.add(mappedAPI);
+                    } catch (APIManagementException e) {
```

Comment:
I think we should not just skip this error and log as debug. This need to be logged as an error because this is an actual error. WDYT?

Reply:
From what I understood from https://github.com/wso2/carbon-apimgt/pull/11372#discussion_r941069575, in this particular case this is not a hindrance to the regular flow i.e., whether the exception gets thrown or not is irrelevant to continuing on with the flow. So maybe more of a warning kinda thing? 🤔

Reply:
Above suggestions is also added to the PR to `9.22.x` branch. (https://github.com/wso2/carbon-apimgt/pull/11375)